### ISSUE TITLE
Configurable auto-advance when step completes

### ISSUE THREAD

**Issue by olman21** [angular/material2#12908]
#### Bug, feature request, or proposal:
Proposal

#### What is the expected behavior?
Optionally allow the consumer define if it wants to auto-advance an step when it completes (by using [completed] step property)

#### What is the current behavior?
Nothing happens as soon the step is marked as completed

#### What is the use-case or motivation for changing an existing behavior?
Use Observable based state such as NgRx stores, and in a more functional way allow set the stepper navigation based on state, now I have to subscribe to store and call navigation via an stepper instance in order to handle steps advance, too imperatively and also results in a awful timing issues because when stepper.next() is called and [completed] property update is on its way nothing happens.

#### Is there anything else we should know?
I'm willing to implement such feature. No breaking changes would be introduced.

**Comment by josephperrott** [comment]
It seems reasonable and if it can be done in a non-breaking way I am sure we would welcome the contribution.

@mmalerba to confirm.

**Comment by mmalerba** [comment]
I assume you mean an output `(completed)` rather than an input `[completed]` correct? That seems like a reasonable feature to me

**Comment by jelbourn** [comment]
Updated this to clarify the feature we would add here: introducing a `completed` event for the stepper/steps so that the application could do whatever it wants upon completion.

**Comment by pkirchniawy** [comment]
Hej there! This would be my first issue but I would love to tackle it.

**Comment by mmalerba** [comment]
Sure, you can add me as a reviewer on your PR when you're ready

**Comment by ramprakashram** [comment]
@mmalerba If this issue is still open, I can take it up ?

**Comment by pkirchniawy** [comment]
Hej @ramprakashram I'm currently working on it :)

**Comment by synopss** [comment]
> I assume you mean an output `(completed)` rather than an input `[completed]` correct? That seems like a reasonable feature to me

I think, the initial request was to use the already existing input `[completed]` value to ask to move to the next step if a new optional property `[auto-advance]` is enabled.

So, if `[completed]` is set to true, then the stepper will automatically move to the next step.

**Comment by jomendez** [comment]
I agree with @synopss, but also believe that the two use cases (exposing an event (completed) and creating an input [auto-avance]) could be very beneficial and shouldn't break existing functionality.

**Comment by Naaajii** [comment]
I've created an PR for this request: [#26817](https://github.com/angular/components/pull/26817). 

This is my first time contributing, please shed light if I have something wrong or missed out something.

**Comment by skilledDev96** [comment]
Is someone working on it? 
@naaajii you deleted your branch?

**Comment by gerbleofriverdale** [comment]
Hi,
Is anybody working on this issue?

**Comment by yousafravian** [comment]
@olman21 @jelbourn @crisbeto 
Can someone add some more clarification to this feature? The original description sounds like it needs an Input property called `completed`, but this [comment](https://github.com/angular/components/issues/12908#issuecomment-634844817) at the same time suggested that `completed` should be an output.

Is there any pending step here? Some more discussions or maybe close this ticket if this is not required

Thank you!

**Comment by noelsebastian22** [comment]
Hi everyone, 

I'm interested in taking over this issue as my first contribution to the project. I see there has been some debate regarding the implementation details.

To ensure I align with the team's current standards for modern Angular, could a maintainer clarify if the preferred approach is:

Adding a (completed) output to CdkStep (as suggested by @mmalerba and @jelbourn).

Adding an [autoAdvance] input to CdkStepper that reacts to changes in the step's completed state.

Both?

Once confirmed, I'll sign the CLA and submit a PR. Is anyone currently active on this?

cc: @mmalerba
